pkgs/applications/networking/cluster/minikube/default.nix

```diff
@@ -1,47 +1,54 @@
-{ stdenv, lib, fetchurl, makeWrapper, docker-machine-kvm, kubernetes, libvirt, qemu }:
+{ stdenv, buildGoPackage, fetchFromGitHub, go-bindata, kubernetes, libvirt, qemu, docker-machine-kvm, makeWrapper }:
 
 let
-  arch = if stdenv.isLinux
-         then "linux-amd64"
-         else "darwin-amd64";
-  checksum = if stdenv.isLinux
-             then "0cdcabsx5l4jbpyj3zzyz5bnzks6wl64bmzdsnk41x92ar5y5yal"
-             else "12f3b7s5lwpvzx4wj6i6h62n4zjshqf206fxxwpwx9kpsdaw6xdi";
+  binPath = [ kubernetes ]
+    ++ stdenv.lib.optionals stdenv.isLinux [ kubernetes libvirt qemu docker-machine-kvm ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [];
 
-# TODO: compile from source
-
-in stdenv.mkDerivation rec {
-  pname = "minikube";
+in buildGoPackage rec {
+  pname   = "minikube";
+  name    = "${pname}-${version}";
   version = "0.17.1";
-  name = "${pname}-${version}";
 
-  src = fetchurl {
-    url = "https://storage.googleapis.com/minikube/releases/v${version}/minikube-${arch}";
-    sha256 = "${checksum}";
-  };
+  goPackagePath = "k8s.io/minikube";
 
-  phases = [ "installPhase" "fixupPhase" ];
+  src = fetchFromGitHub {
+    owner  = "kubernetes";
+    repo   = "minikube";
+    rev    = "v${version}";
+    sha256 = "1m61yipn0p3cfavjddhrg1rcmr0hv6k3zxvqqd9fisl79g0sdfsr";
+  };
 
-  buildInputs = [ makeWrapper ];
+  buildInputs = [ go-bindata makeWrapper ];
+  subPackages = [ "cmd/minikube" "cmd/localkube" ];
 
-  binPath = lib.makeBinPath [ docker-machine-kvm kubernetes libvirt qemu ];
+  preBuild = ''(
+    cd go/src/${goPackagePath}
+    go-bindata -nomemcopy -o pkg/minikube/assets/assets.go -pkg assets -prefix deploy/addons deploy/addons/...
+  )'';
 
-  installPhase = ''
-    install -Dm755 ${src} $out/bin/${pname}
+  buildFlagsArray = ''
+    -ldflags=
+      -X k8s.io/minikube/pkg/version.version=${version}
+      -X k8s.io/minikube/pkg/version.isoVersion=v1.0.7
+      -X k8s.io/minikube/pkg/version.isoPath=minikube/iso
   '';
 
-  fixupPhase = ''
-    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" "$out/bin/minikube"
-
-    wrapProgram $out/bin/${pname} \
-      --prefix PATH : ${binPath}
+  # TODO: turn these completions on (this code seemed to fail)
+  postInstall = ''
+    # mkdir -p $bin/share/bash-completion/completions/
+    # HOME=$PWD $bin/bin/minikube completion bash > $bin/share/bash-completion/completions/minikube
+  '' + stdenv.lib.optionalString stdenv.isDarwin ''
+    install_name_tool -delete_rpath $out/lib $bin/bin/localkube
   '';
```

Comment:
That seems like a more general `buildGoPackage` thing, right? Also, the `bin` output doesn't have the source.